vitest.config.ts

```diff
@@ -3,13 +3,13 @@ import { defineConfig } from 'vitest/config';
 export default defineConfig({
   resolve: {},
   test: {
-    pool: 'threads',
+    pool: 'vmThreads',
```

Comment:
figured that this is even faster, at least on my machine
tests run in 4s for me with this PR
there are now around 753 tests in vitest

Reply:
i think there was some kind of issue in the first PR that let me choose threads over vmThreads.
I looked into it, i don't really remember.
Let's see what happen

Reply:
I think we need to move this earlier to vitest phase
https://github.com/fabricjs/fabric.js/blob/master/.github/workflows/tests.yml#L134
because browser tests in vitest are using playwright as a provider

Reply:
> i think there was some kind of issue in the first PR that let me choose threads over vmThreads. I looked into it, i don't really remember. Let's see what happen

first PR by default used forks if I am not mistaken
https://vitest.dev/config/#pool
and that was really slow

Reply:
I need to review and check a bunch of things.
RUnning tests in browser is something that needs to be discussed.
Today we do that with qunit and testem.
The unit tests at the of the game needs to run once per supported node version and at least once in chrome. All of them. Not just some and some.

So i need to understand this browser thing a little bit better

Reply:
Browser thing in vitest is just another environment option, there is not much to it
There are few of them:
JSdom is node implementation of the simulate browser env, which doesn't work for few of the tests like for setting styles where cssText is mutated
JSDom doesn't handle that, it is missing implementation, so I just made test to run in real browser environment.
All other tests still run in jsdom.
There is also a happydom environmet which is a faster version of jsdom but a bit more barebones.
And there is node environment but there is no concept of document and window in it.

So In this PR I just made two tests to run in browser which are image loading because it doesn't work in jsdom, and setting of styles via mutation of cssText which also doesn't work in jsdom.

Reply:
I think I understand what you mean
you want to run all tests multiple times
once in chromium
once in jsdom

we can do that easily in vitest by leveraging projects
the same way I added the browser configuration we can easily add the new entry into it that will run all tests in browser mode as well
currently all unit tests are ran in vitest only with jsdom, which is the exact same thing when you were using jest

Reply:
ok let's finish to convert tests from qunit and let's skip the tests that requires browser for now, or let's see if they can be fixed however they were fixed in qunit, but using a skip + message is fine too.

When qunit and testm are gone, we can write a github action to run witest in browser without having the same one running both serially.

I don't want to mix the two things for now becuase it becomes an overhead for me following everything, is just a matter of organization for me

Reply:
ok sounds good
i will skip the 2 tests for now and will remove the browser settings for vitest
after whole migration is done it will be easy to just add new vitest config for browser mode that can be run separately

Reply:
ok that should be it
I skipped the two tests and added TODO explaining the error

Reply:
in this PR https://github.com/fabricjs/fabric.js/pull/10521
I figured out why images weren't loading in jsdom and I fixed it
I can unskip the image test in this PR after other one is merged